wdl-lint/src/rules/malformed_lint_directive.rs

```diff
@@ -0,0 +1,111 @@
+use crate::TagSet;
+use crate::util::is_inline_comment;
+
+/// The identifier for the Malformed Lint Directive rule.
+const ID: &str = "MalformedLintDirective";
+/// The accepted lint directives.
+const ACCEPTED_LINT_DIRECTIVES: [&str; 1] = ["except"];
+
+/// Creates an "Inline Lint Directive" diagnostic.
+fn inline_lint_directive(
+    span: Span,
+) -> Diagnostic {
+    Diagnostic::warning(
+        "lint directive must be on its own line")
+        .with_rule(ID)
+        .with_label("malformed lint directive", span,
+        )
+        .with_fix("place lint directive on new line")
+}
+
+/// Creates an "Invalid Lint Directive" diagnostic.
+fn invalid_lint_directive(
+    span: Span,
+) -> Diagnostic {
+    let accepted_directives = ACCEPTED_LINT_DIRECTIVES.join(", ");
+    Diagnostic::warning(
+        "lint directive not recognized")
+        .with_rule(ID)
+        .with_label("lint directive not recognized", span)
+        .with_fix(format!("remove unrecognized lint directives, use any of the directives included: [{:#?}]", accepted_directives))
+}
+
+fn no_colon_detected(span: Span) -> Diagnostic {
+    Diagnostic::warning(
+        "no colon detected following lint directive")
+        .with_rule(ID)
+        .with_label("no colon detected following lint directive", span)
+        .with_fix("add colon following lint directive")
+}
+
+/// Detects a malformed lint directive.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct MalformedLintDirectiveRule;
+
+impl Rule for MalformedLintDirectiveRule {
+    fn id(&self) -> &'static str {
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Flags malformed lint directives"
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Comments which begin with `#@` must only contain valid lint directives. Lint directives \
+        must be on their own line, only preceded by whitespace."
+    }
+
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::Clarity, Tag::Correctness])
+    }
+
+    fn exceptable_nodes(&self) -> Option<&'static [wdl_ast::SyntaxKind]> {
+        // Some(&[SyntaxKind::VersionStatementNode])
+        None
+    }
+}
+
+impl Visitor for MalformedLintDirectiveRule {
+    type State = Diagnostics;
+
+    fn document(&mut self, _: &mut Self::State, _: VisitReason, _: &Document, _: SupportedVersion) {
+        // This is intentionally empty, as this rule has no state.
+    }
+
+    fn comment(&mut self, state: &mut Self::State, comment: &Comment) {
+        if let Some(lint_directive) = comment.as_str().strip_prefix("#@") {
+            if is_inline_comment(comment) {
+                state.add(inline_lint_directive(
+                    comment.span(),
+                ));
+            }
+
+            let mut directive = lint_directive.trim().split(" ").next().expect("directive");
+            if !directive.ends_with(":") {
+                state.add(no_colon_detected(
+                    Span::new(3 + directive.chars().count(), 1),
```

Comment:
Here's your problem. With `Span::new(3 + directive.chars().count(), 1)` you are creating a new span pointing at the 3rd character _in the entire document_. I imagine you probably wanted to start the Span 3 characters past the start of the comment? That would be something like:
```suggestion
                    Span::new(comment.span().start() + 3 + directive.chars().count(), 1),
```